jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -242,6 +242,10 @@ def _parse_hdulist(app, hdulist, file_name=None,
         # to sky regions, where the parent data of the subset might have dropped spatial WCS info
         metadata['_orig_spatial_wcs'] = _get_celestial_wcs(wcs)
 
+        # metadata needs to be added before data is added to data_collection
+        if data_type == 'flux' or data_type == 'uncert':
```

Comment:
Is there any reason we can't just always set this (i.e remove the if-statement entirely)?  If not, this (and other appearances below) can be simplified to `if data_type in ('flux', 'uncert'):`.

Reply:
removing the ifs in the parsers will work if I account for 'mask' being assigned as metadata, so:
for data in self.app.data_collection:
            if data.meta.get('_cube_data_type') **and data.meta.get('_cube_data_type') != 'mask'**:

Reply:
why is that necessary to add?  Neither the if or the elif will evaluate to true anyways, right?

Reply:
It was something in the commit you had initially commented on that was causing the subset tests to fail with get_subset_object, but I can't recreate that, all the tests are passing and the the behavior is correct in the gui so I'll push that change